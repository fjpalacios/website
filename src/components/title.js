import * as React from 'react'
import './title.scss'

const Title = ({ title }) => {
  return (
    <div className="title">
      <h3>{title}</h3>
    </div>
  )
}

export default Title
