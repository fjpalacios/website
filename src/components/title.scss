.title {
  margin-bottom: 20px;
  padding: 10px 20px;
  transition: all .45s ease;
  background-color: $primary;
  color: $inverted-text;
  line-height: 0;
  text-transform: uppercase;

  @include print {
    margin-bottom: 10px;
    padding: 1px 10px;
  }

  h3 {
    margin: 17px 0 13px;
    font-weight: 100;

    @include print {
      margin: 10px 0 7px;
    }
  }
}
