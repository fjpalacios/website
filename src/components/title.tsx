import './title.scss'
import React, { FunctionComponent, ReactElement } from 'react'

type titleProps = {
  title: string
}

export const Title: FunctionComponent<titleProps> = ({
  title,
}): ReactElement => {
  return (
    <div className="title">
      <h2>{title}</h2>
    </div>
  )
}
