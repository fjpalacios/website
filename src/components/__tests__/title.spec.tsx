import '@testing-library/jest-dom'
import createComponentWithIntl from '../../../__helpers__/i18n-component'
import React from 'react'
import { Title } from '../title'

describe('Title', () => {
  it('should render properly', () => {
    const title = createComponentWithIntl(<Title title="Foo" />)
    expect(title).toMatchSnapshot()
  })

  it('should show the passed prop as a title', () => {
    const { container } = createComponentWithIntl(<Title title="Foo" />)
    const title = container.getElementsByClassName('title')[0].textContent
    expect(title).toBe('Foo')
  })
})
