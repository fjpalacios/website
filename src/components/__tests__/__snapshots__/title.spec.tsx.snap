// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render properly 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={
    Object {
      "about": Object {
        "education": "Education",
        "experience": "Experience",
        "freelance": "Freelance work",
        "me": "About me",
        "projects": "Side projects",
        "talks": "Talks / workshops",
        "volunteering": "Volunteering",
      },
      "menu": Object {
        "switchTheme": "Switch Theme",
        "toEnglish": "Switch to English language",
        "toSpanish": "Switch to Spanish language",
      },
      "metaData": Object {
        "description": "Francisco Javier Palacios Pérez's personal website, Software Developer based in Valencia (Spain)",
        "image": "https://fjp.es/logo.jpg",
        "twitterUsername": "@fjpalacios",
        "url": "https://fjp.es",
      },
      "subtitle": "Software Developer",
      "title": "Francisco Javier Palacios Pérez",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <Title
    title="Foo"
  >
    <div
      className="title"
    >
      <h3>
        Foo
      </h3>
    </div>
  </Title>
</IntlProvider>
`;
